java/Mekhmet-Efe_Adzhar/src/com/zoolatech/lecture2/tasks/_1/Number1.java

```diff
@@ -0,0 +1,154 @@
+
+/*Define a class that represents a calculator.
+ *  The class should provide methods that accept another value and perform
+ *  addition, subtraction, multiplication and division operations
+ *  on a value stored in a calculator instance.
+ *  Divides by a zero should be forbidden and ignored.
+ *  The class should also provide a method to get a current value.
+ *  The class should work with both integer and double numbers (ignore roundoff errors).
+ *  Assume all operation results fit into the range of values for a current value type.
+ */
+
+public class Number1 {
+    static Scanner scanner = new Scanner(System.in);
+     static Calculator calculator = new Calculator(0);
+
+    public static void main(String[] args) {
+
+        System.out.println("1. Start");
+        Scanner scanner = new Scanner(System.in);
+
+        if (scanner.nextInt() == 1) {
+            calculations();
+        } else {
+            throw new IllegalArgumentException("Unknown" + scanner.nextInt());
+        }
+    }
+
+    public static void calculations() {
+        System.out.println("Input first number:");
+        if (scanner.hasNextInt()) {
+            integerCalc();
+        } else if (scanner.hasNextDouble()) {
+            doubleCalc();
+        }
+    }
+
+    private static void integerCalc() {
+        calculator = new Calculator(scanner.nextInt());
+        calculator.setInteger(true);
+        System.out.println("Input second integer");
+        int numberTwo = scanner.nextInt();
+        char operation = scanner.next().charAt(0);
+        switch (operation) {
+            case '+' -> calculator.addition(numberTwo);
+            case '-' -> calculator.subtraction(numberTwo);
+            case '*' -> calculator.multiplication(numberTwo);
+            case '/' -> calculator.division(numberTwo);
+            default -> System.out.println("Error");
+        }
+    }
+
+    private static void doubleCalc() {
+        calculator = new Calculator(scanner.nextDouble());
+        calculator.setInteger(false);
+        System.out.println("Input second double");
+        double numberTwoDouble = scanner.nextDouble();
+        char operation = scanner.next().charAt(0);
+        switch (operation) {
+            case '+' -> calculator.addition(numberTwoDouble);
+            case '-' -> calculator.subtraction(numberTwoDouble);
+            case '*' -> calculator.multiplication(numberTwoDouble);
+            case '/' -> calculator.division(numberTwoDouble);
+            default -> System.out.println("ERROR");
+        }
+    }
+}
+
+
+class Calculator {
+
+    private boolean isInteger;
+    private final double numberOne;
+
+    public double getNumberOne() {
+        return numberOne;
+    }
+
+    public Calculator(double numberOne) {
+        this.numberOne = numberOne;
+    }
+
+    public void setInteger(boolean integer) {
+        isInteger = integer;
+    }
+
+    public void addition(double numberTwoDouble) {
+        if (isInteger) {
```

Comment:
If we accept and store values as double numbers, why do we need to have this `isInteger` variable and conditions? we can simply perform all operations assuming using the double type